lib/Tools/LinearLayout.cpp

```diff
@@ -165,6 +165,35 @@ void assertDimsSubsetIgnoringOrder(T &&small, U &&big) {
   }
 }
 
+template <typename T, typename U>
+void assertCommonDimsSameOrder(T &&outerDims, U &&innerDims) {
+  // Check that elements common to both outerDimsRange and innerDimsRange
+  // appear in the same relative order.
```

Comment:
This comment describes the behavior of the function.  Therefore move it outside the function?